lib/authentication.rb

```diff
@@ -0,0 +1,15 @@
+require "jwt"
```

Comment:
Tại sao lại cần jwt anh nhỉ, anh thử nói con đường hoạt động của jwt trong API đi :>

Reply:
Đầu tiên người dùng đăng nhập thì nó sẽ tạo jwt trong lib/authentication nó sẽ gọi tới hàm encode. JWT sẽ encode và decode theo thuật toán hash mà ta muốn dùng (ở đây mình dùng HS256) và một đoạn string mã hóa mà mình tự tạo ra. 
Sau đó nó gửi jwt xuống client tại client jwt sẽ được lưu vào cookies và gửi kèm lên server thông qua header của request. 
Trên server nhận request lấy ra jwt token rồi decode nó trong lib/authentication lấy ra user_id và tìm kiếm user trong database nếu mà user_id (tìm thấy người dùng trong db) thì xác thực thành công và tiếp tục thực hiện request. Nếu không nó sẽ gửi trả về lỗi hay là làm gì đó do mình định nghĩa.